tests/test_cms_menus.py

```diff
@@ -1495,3 +1543,188 @@ def test_menu_with_multiple_languages(self):
         ]
 
         self.assertTreeQuality(context_it['children'], mock_it_tree, 'title')
+
+
+class GetMainNavigationTestCase(CMSTestCase):
+
+    def setUp(self):
+        self.request = RequestFactory().get("/")
+        self.user = factories.UserFactory()
+        self.request.user = self.user
+        self.renderer = menu_pool.get_renderer(self.request)
+        self.menu = CMSMenu(self.renderer)
+
+    def test_main_navigation_found(self):
+        """
+        When passed a queryset of multiple MenuContent objects, and one is related to a Menu that is marked as the main
+        navigation, the returned queryset should only contain the MenuContent related to the main navigation
+        """
+        not_main_navigation = factories.MenuContentFactory(language="en", menu__main_navigation=False)
+        main_navigation = factories.MenuContentFactory(language="en", menu__main_navigation=True)
+        all_menucontents = MenuContent._base_manager.all()
+
+        result = self.menu.get_main_navigation(menucontents=all_menucontents)
+
+        self.assertEqual(result.count(), 1)
+        self.assertIn(main_navigation, result)
+        self.assertNotIn(not_main_navigation, result)
+
+    def test_main_navigation_not_found(self):
+        """
+        When none of the MenuContent objects are related to a Menu marked as the main navigation, the queryset is
+        returned unchanged
+        """
+        factories.MenuContentFactory.create_batch(2, language="en", menu__main_navigation=False)
+        all_menucontents = MenuContent._base_manager.all()
+
+        result = self.menu.get_main_navigation(menucontents=all_menucontents)
+
+        self.assertEqual(result, all_menucontents)
+        self.assertEqual(all_menucontents.count(), result.count())
+
+    def test_multiple_main_navigation(self):
+        """
+        When multiple MenuContent objects are related to a Menu marked as the main navigation are found, all the
+        MenuContent objects should be returned
+        """
+        older = factories.MenuContentWithVersionFactory(language="en", menu__main_navigation=True)
```

Comment:
Perhaps a bit pedantic, but could we change these to older_menucontent etc. Although it's a simple test, in general it's more readable to be a little more verbose.